rest-service/manager_rest/blueprints_manager.py

```diff
@@ -143,16 +143,21 @@ def update_snapshot_status(self, snapshot_id, status, error):
     def create_snapshot(self, snapshot_id,
                         include_metrics, include_credentials):
         self.create_snapshot_model(snapshot_id)
-        _, execution = self._execute_system_workflow(
-            wf_id='create_snapshot',
-            task_mapping='cloudify_system_workflows.snapshot.create',
-            execution_parameters={
-                'snapshot_id': snapshot_id,
-                'include_metrics': include_metrics,
-                'include_credentials': include_credentials,
-                'config': self._get_conf_for_snapshots_wf()
-            }
-        )
+        try:
+            _, execution = self._execute_system_workflow(
+                wf_id='create_snapshot',
+                task_mapping='cloudify_system_workflows.snapshot.create',
+                execution_parameters={
+                    'snapshot_id': snapshot_id,
+                    'include_metrics': include_metrics,
+                    'include_credentials': include_credentials,
+                    'config': self._get_conf_for_snapshots_wf()
+                }
+            )
+        except manager_exceptions.ExistingRunningExecutionError:
```

Comment:
Maybe catch any exception here? If another exception gets raised, it would be also a good idea to clean up the snapshot (?)